crates/core/src/table/mod.rs

```diff
@@ -291,23 +292,76 @@ impl DeltaTable {
         &self,
         filters: &[PartitionFilter],
     ) -> Result<Vec<String>, DeltaTableError> {
-        let files = self.get_files_by_partitions(filters).await?;
-        Ok(files
-            .iter()
-            .map(|fname| self.log_store.to_uri(fname))
-            .collect())
+        let Some(state) = self.state.as_ref() else {
+            return Err(DeltaTableError::NotInitialized);
+        };
+        let mut out: Vec<String> = Vec::new();
+        let mut stream = state
+            .snapshot()
+            .file_views_by_partitions(&self.log_store, filters);
+        while let Some(item) = stream.next().await {
+            let add = item?;
+            let p_cow = add.path();
+            let p = p_cow.as_ref();
+            let obj_path = add.object_store_path();
+            out.push(self.resolve_add_path_to_uri(p, &obj_path));
+        }
+        Ok(out)
     }
 
     /// Returns a URIs for all active files present in the current table version.
     pub fn get_file_uris(&self) -> DeltaResult<impl Iterator<Item = String> + '_> {
-        Ok(self
-            .state
-            .as_ref()
-            .ok_or(DeltaTableError::NotInitialized)?
-            .log_data()
-            .into_iter()
-            .map(|add| add.object_store_path())
-            .map(|path| self.log_store.to_uri(&path)))
+        let state = self.state.as_ref().ok_or(DeltaTableError::NotInitialized)?;
+        let root = self.log_store.config().location.clone();
+        // The clone is necessary here because the closure needs to own `root` (a `Url`),
+        // as it is moved into the closure for the iterator, which may outlive the current scope.
+        #[allow(clippy::redundant_clone)]
+        Ok(state.log_data().into_iter().map(move |add| {
+            let p_cow = add.path();
+            let p = p_cow.as_ref();
+            // Try parsing as a URI first. If it parses, it's an absolute URI.
+            // If it fails with RelativeUrlWithoutBase (or any other error),
+            // treat it as a filesystem path and check if it's absolute.
+            let obj_path = add.object_store_path();
+            // We still keep `root` cloned above to preserve the move semantics in iterator,
+            // but the resolution logic is centralized in `resolve_add_path_to_uri`.
+            let _ = &root; // keep root captured in the closure
+            self.resolve_add_path_to_uri(p, &obj_path)
+        }))
```

Comment:
This looks quite odd to me. Why is root captured but not used? The closure already captures a reference to `self` so I don't think explicitly capturing root is necessary and this can all be simplified:
```suggestion
        Ok(state.log_data().into_iter().map(move |add| {
            let obj_path = add.object_store_path();
            self.resolve_add_path_to_uri(&add.path(), &obj_path)
        }))
```